mangum/adapter.py

```diff
@@ -91,89 +91,108 @@ def __call__(self, event: dict, context: "LambdaContext") -> dict:
                 )
                 stack.enter_context(lifespan_cycle)
 
-            request_context = event["requestContext"]
+            scope = self.event_to_asgi_scope(event)
+            scope["aws.event"] = event
+            scope["aws.context"] = context
+            http_cycle = HTTPCycle(scope, text_mime_types=self.text_mime_types)
 
-            if event.get("multiValueHeaders"):
-                headers = {
-                    k.lower(): ", ".join(v) if isinstance(v, list) else ""
-                    for k, v in event.get("multiValueHeaders", {}).items()
-                }
-            elif event.get("headers"):
-                headers = {k.lower(): v for k, v in event.get("headers", {}).items()}
-            else:
-                headers = {}
+            initial_body = self.event_to_request_body(event)
+            response = http_cycle(self.app, initial_body)
 
-            # API Gateway v2
-            if event.get("version") == "2.0":
-                source_ip = request_context["http"]["sourceIp"]
-                path = request_context["http"]["path"]
-                http_method = request_context["http"]["method"]
-                query_string = event.get("rawQueryString", "").encode()
+        return response
 
-                if event.get("cookies"):
-                    headers["cookie"] = "; ".join(event.get("cookies", []))
+    def event_to_request_body(self, event: dict) -> bytes:
```

Comment:
Remove this method and keep the logic in the main `__call__` function for now.

Reply:
done